src/lib/libmints/deriv.cc

```diff
@@ -595,31 +595,34 @@ SharedMatrix Deriv::compute()
                terms below are computed correctly.  The two-particle terms are computed the same in both cases
                as all spin cases have been collapsed into the a single SO TPDM. */
 
-            // Dial up an integral tranformation object to backtransform the OPDM, TPDM and Lagrangian
-            vector<boost::shared_ptr<MOSpace> > spaces;
-            spaces.push_back(MOSpace::all);
-            boost::shared_ptr<IntegralTransform> ints_transform = boost::shared_ptr<IntegralTransform>(
-                        new IntegralTransform(wfn_,
-                                              spaces,
-                                              wfn_->same_a_b_orbs() ? IntegralTransform::Restricted : IntegralTransform::Unrestricted, // Transformation type
-                                              IntegralTransform::DPDOnly,    // Output buffer
-                                              IntegralTransform::QTOrder,    // MO ordering
-                                              IntegralTransform::None));     // Frozen orbitals?
-            dpd_set_default(ints_transform->get_dpd_id());
-
-            // Some codes already presort the tpdm, do not follow this as an example
-            if(Process::environment.options.get_bool("DERIV_TPDM_PRESORTED"))
-                ints_transform->set_tpdm_already_presorted(true);
-
-            ints_transform->backtransform_density();
-
-            Da = factory_->create_shared_matrix("SO-basis OPDM");
-            Db = factory_->create_shared_matrix("NULL");
-            Da->load(_default_psio_lib_, PSIF_AO_OPDM);
-            X = factory_->create_shared_matrix("SO-basis Lagrangian");
-            X->load(_default_psio_lib_, PSIF_AO_OPDM);
-            // The CC lagrangian is defined with a different prefactor to SCF / MP2, so we account for it here
-            X->scale(0.5);
+            if ( !Process::environment.options.get_bool("DERIV_DENSITY_BACKTRANSFORMED") ) {
```

Comment:
As a general comment: anyone else think `Deriv` should only have class options with setters and not global options?